instrumentation/httpasyncclient/src/test/java/brave/httpasyncclient/ITTracingHttpAsyncClientBuilder.java

```diff
@@ -92,4 +99,52 @@ protected void post(CloseableHttpAsyncClient client, String pathIncludingQuery,
     
     takeSpan();
   }
+
+  @Test public void currentTraceContextIsVisibleInCallbackThread() throws Exception {
+    Tracer tracer = httpTracing.tracing().tracer();
+    AtomicBoolean callbackHasTraceCtx = new AtomicBoolean(false);
+    AtomicBoolean callbackCompleted = new AtomicBoolean(false);
+    server.enqueue(new MockResponse());
+    closeClient(client);
```

Comment:
is restarting the client needed here? what happens if you don't close it? it looks like it would be the same as what you're doing. maybe try without lines 108-110 or add a comment above why we have to recreate it.